src/test/scala/org/broadinstitute/dsde/firecloud/integrationtest/ElasticSearchShareLogDAOSpec.scala

```diff
@@ -0,0 +1,107 @@
+package org.broadinstitute.dsde.firecloud.integrationtest
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.firecloud.integrationtest.ESIntegrationSupport.{searchDAO, shareLogDAO}
+import org.broadinstitute.dsde.firecloud.model.ShareLog.{Share, ShareType}
+import org.scalatest.{BeforeAndAfterAll, FreeSpec, Matchers}
+
+import scala.util.Try
+
+class ElasticSearchShareLogDAOSpec extends FreeSpec with Matchers with BeforeAndAfterAll with LazyLogging {
+  override def beforeAll = {
+    // using the recreate from search dao because we don't have recreate in sharelog dao
+    searchDAO.recreateIndex()
+    ElasticSearchShareLogDAOSpecFixtures.fixtureShares map { share =>
+      shareLogDAO.logShare(share.userId, share.sharee, share.shareType)
+    }
+  }
+
+  override def afterAll = {
+    // using the delete from search dao because we don't have recreate in sharelog dao
+    searchDAO.deleteIndex()
+  }
+
+  "ElasticSearchShareLogDAO" - {
+    "logShare" - {
+      "should log a share and get it back successfully using the generated MD5 hash" in {
+        val share = Share("roger", "syd@gmail.com", ShareType.WORKSPACE)
+        val loggedShare = shareLogDAO.logShare(share.userId, share.sharee, share.shareType)
+        val check = shareLogDAO.getShare(share)
+        assertResult(loggedShare) { check }
+      }
+      "should successfully log a duplicate share" in {
+        val loggedShare = shareLogDAO.logShare("fake4", "fake3@gmail.com", ShareType.WORKSPACE)
+        val check = Try(shareLogDAO.logShare(loggedShare.userId, loggedShare.sharee, loggedShare.shareType))
+        assert(check.isSuccess)
+      }
```

Comment:
how is this a duplicate?

Reply:
Bad wording! Changing to `"should successfully log a record of a user sharing a workspace with the same user twice"`